Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py

```diff
@@ -0,0 +1,687 @@
+            method="GET",
+            url_suffix="/reportpod/logs/sync",
+            params=params,
+            resp_type="response",
+            stream=True,
+        )
+
+
+""" HELPER FUNCTIONS """
+
+
+def get_fetch_interval(fetch_interval: str | None) -> int:
+    """Returns the fetch interval in seconds"""
+    fetch_sleep = arg_to_number(fetch_interval)
+    if not fetch_sleep:
+        return DEFAULT_FETCH_SLEEP
+    if fetch_sleep < DEFAULT_FETCH_SLEEP:
+        demisto.debug(
+            f"Fetch interval is too low, setting it to minimum of {DEFAULT_FETCH_SLEEP} seconds"
+        )
+        return DEFAULT_FETCH_SLEEP
+    return fetch_sleep
+
+
+def get_events_and_write_to_file_system(
+    client: Client,
+    params: dict,
+) -> Path:
+    """
+    Writing the events that come from the API to a temporary file.
+    Return:
+        Path: the file path
+    """
+    with client.get_logs(params) as res, tempfile.NamedTemporaryFile(
+        mode="wb", delete=False
+    ) as tmp_file:
+        # Write the chunks from the response to the tmp file
+        for chunk in res.iter_content(chunk_size=MAX_CHUNK_SIZE_TO_WRITE):
+            tmp_file.write(chunk)
+
+    return Path(tmp_file.name)
+
+
+def get_start_and_end_date(start_date: str | None) -> tuple[int, int]:
+    """
+    returns `start_date` and `end_date`
+
+    Args:
+        start_date (str | None): start_date which is stored in the last_run object from the second run onwards
+
+    Returns:
+        tuple[int, int]: start_date, end_date
+    """
+    # set the end_date to the current time
+    now = datetime.now().astimezone(pytz.utc)
+
+    # If there is no `start_date` stored in the `last_run` object,
+    # sets the `start_date` to one minute before the current time
+    start_date = int(start_date or date_to_timestamp(now - timedelta(minutes=1)))
+
+    # convert the end_date to timestamp
+    end_date = date_to_timestamp(date_str_or_dt=now)
+
+    return start_date, end_date
+
+
+def get_status_and_token_from_file_system(file_path: Path) -> tuple[str, str]:
+    """
+    Extracting the status and the next_token.
+    """
+
+    # Getting the file size to read only its end for the `status` and `next_token`
+    file_size = get_file_size(file_path)
+    read_size = 2000
+    if file_size < read_size:  # In case the file is smaller than 2000 bytes
+        read_size = file_size
+
+    # Reading end of file
+    with file_path.open("rb") as tmp_file:
+        tmp_file.seek(file_size - read_size)
+        end_file = tmp_file.read()
+
+    # Extracting the `status` and the `next_token` by regex
+    status = ""
+    token = ""
+    if status_match := REGEX_FOR_STATUS.search(str(end_file)):
+        status = status_match.groupdict().get("status", "")
+    if token_match := REGEX_FOR_TOKEN.search(str(end_file)):
+        token = token_match.groupdict().get("token", "")
+
+    return status, token
+
+
+def get_file_size(file_path: Path) -> int:
```

Comment:
this is not really nesseary.
better use stat().st_size directly/

also when you write something like this(and you did):
file_size = get_file_size(file_path)
you already know form the variable name that it is file size.
so this is just as readable:
file_size = file_path.stat().st_size.